python/taichi/lang/matrix.py

```diff
@@ -415,13 +415,19 @@ def E(x, y):
 
     @staticmethod
     def cross(a, b):
-        assert a.n == 3 and a.m == 1
-        assert b.n == 3 and b.m == 1
-        return Matrix([
-            a(1) * b(2) - a(2) * b(1),
-            a(2) * b(0) - a(0) * b(2),
-            a(0) * b(1) - a(1) * b(0),
-        ])
+        if a.n == 3 and a.m == 1 and b.n == 3 and b.m == 1:
+            return Matrix([
+                a(1) * b(2) - a(2) * b(1),
+                a(2) * b(0) - a(0) * b(2),
+                a(0) * b(1) - a(1) * b(0),
+            ])
+
+        elif a.n == 2 and a.m == 1 and b.n == 2 and b.m == 1:
+            return Matrix(a(0) * b(1) - a(1) * b(0))
```

Comment:
```suggestion
            return a(0) * b(1) - a(1) * b(0)
```
We want to return scalar value instead of 1x1 matrices.